### PR TITLE
Update @sap/cds-dk to 6.8.1

### PR DESCRIPTION
**Pull request by beckermarc**
Aligned MTA deployment files and MTX Sidecar with best practices Introduced workspaces concept for mtx/sidecar to ensure availability of @sap/cds-mtxs in root node_modules

### CODE REVIEW

**Comment by beckermarc** [comment]
Note, that I will do the switch to npm ci and check in of package-lock.json in a separate PR, as I want to do some more tests with that.

**Comment by beckermarc** [comment]
I think we still have an issue with version locking in that approach: It seems like the open version ranges from mtx/sidecar/package.json now propagate to our installation of @sap/cds-dk. Even if I have a package-lock.json in mtx/sidecar that pins to e.g. cds 6.7.2 and I install cds-dk 6.7.2 with npm install --no-save I get cds 6.8.1 in the node_modules. Without the workspace configuration this isn't the case (+ no flat node_modules).

I haven't tested it, but I suspect that this may be fixed, if package-lock.json was generated into the root folder. But there it is currently not picked up by the cds build when copying over to the gen folder. I think we will need to revisit that approach and find a solution for that problem.

**Review by swaldmann — APPROVED** [review]
Nice 👍

File: mtx/sidecar/package.json [hunk 1]
```diff
@@ -30,8 +14,11 @@
   "engines": {
     "node": "^16"
   },
+  "cds": {
+    "profile": "mtx-sidecar"
+  },
   "scripts": {
     "start": "cds-serve",
-    "build": "npm install && npx cds build ../.. --for mtx-sidecar && cd gen && npm install"
+    "build": "cds build ../.. --for mtx-sidecar --production && cd gen && npm install"
```
Comment (LotharBender) [inline mtx/sidecar/package.json:22]: Productive applications should go for the package-lock approach in order to ensure reproducible deployments

File: srv/src/main/resources/application.yaml [hunk 2]
```diff
@@ -99,7 +99,7 @@ cds:
 spring:
   config.activate.on-profile: ft
 cds:
-  model.provider.url: http://localhost:4004
+  model.provider.url: http://localhost:4005
```
[the remark this answers is not in the document]
Comment (beckermarc) [reply srv/src/main/resources/application.yaml:102]: Apparently...didn't find any note on that so far, but seems to be the case.

**Comment by LotharBender** [comment]
> I think we still have an issue with version locking in that approach: It seems like the open version ranges from...

I cannot reproduce this behaviour - with node@18. 
npm is consistently installing the cds-mtxs and cds versions defined in 'mtx/sidecar/package-lock.json` in the `project/node_modules` folder. These are the versions that are used later on  by `cds build` when resolving cds models .  The cds-dk dependencies are installed in a nested way in case their version is different from the versions defined by the package-lock.json located in the sidecar folder.
Thus, I do not see any conflict.

@beckermarc - can you check again with a package-lock.json located in the sidecar folder?

**Comment by beckermarc** [comment]
Removed the workspace concept from this PR and will merge this PR to unblock the usage of @sap/cds-dk 6.8.0, profile settings and mta.yaml updates.. I'll open a follow-up PR for the workspace concept again.

**Review by mofterdinger — APPROVED** [review]

**Pull request merged**